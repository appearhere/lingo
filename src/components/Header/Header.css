.root {
  display: grid;
  grid-template-columns: 0.2fr 0.4fr 0.4fr;
  grid-template-areas:
  "logo logo logo"
  "search search search"
  "navigation navigation navigation";
  align-items: center;
  position: relative;
  background-color: var(--color-white);
  grid-gap: var(--size-regular);
  padding-top: var(--size-regular);
  padding-bottom: var(--size-regular);
  padding-left: calc(var(--size-lg-i) + var(--size-sm-ii));
  padding-right: calc(var(--size-lg-i) + var(--size-sm-ii));
  min-height: 0;
  min-width: 0;
  box-shadow: 0 0 5px 0 rgba(0,0,0,.1);
}

.logoContainer {
  padding: var(--size-regular);
  padding-left: 0;
  grid-area: logo;
  position: relative;
  display: flex;
  align-items: center;
  text-decoration: none;
  overflow: hidden;
  min-width: 0;
  min-height: 0;
}

.logo {
  color: var(--color-black);
  font-size: var(--fontsize-large-i);
  margin-right: var(--size-sm-i);
}

.logoText {
  color: var(--color-black);
  font-weight: var(--fontweight-demi);
  font-size: var(--fontsize-large-i);
}

.navigation {
  grid-area: navigation;
  overflow-x: auto;
  white-space: nowrap;
  min-width: 0;
  min-height: 0;
}

.navigationItem {
  font-family: inherit;
  padding: var(--size-regular);
  color: var(--color-grey);
}

.navigationItem:first-of-type {
  padding-left: 0;
}

.navigationItem:hover {
  color: var(--color-black);
}

.navigationItem:focus {
  outline: none;
}

.active {
  color: var(--color-black);
}

@media(min-width: 36.25rem) {
  .root {
    padding-top: calc(var(--size-lg-i) + var(--size-sm-ii));
    padding-bottom: calc(var(--size-lg-i) + var(--size-sm-ii));
    padding-left: var(--size-lg-iii);
    padding-right: var(--size-lg-iii);
  }
}

@media(min-width: 70rem) {
  .root {
    grid-template-columns: 0.1fr 0.5fr 0.4fr;
    grid-template-areas:
    "logo navigation search";
  }
}
