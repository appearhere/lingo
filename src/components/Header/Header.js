import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { subscribe } from 'subscribe-ui-event';

import TabBar from '@appearhere/bloom/components/Navigation/TabBar/TabBar';
import TabBarItem from '@appearhere/bloom/components/Navigation/TabBar/TabBarItem';

import m from '@appearhere/bloom/globals/modifiers.css';

const MIN_MOBILE_NAV_WIDTH = '(min-width: 36.25rem)';

export default class Header extends Component {
  static propTypes = {
    match: PropTypes.shape({}).isRequired,
    history: PropTypes.shape({}).isRequired,
  };

  state = {
    isNavScrollable: false,
  };

  componentDidMount() {
    this.checkWindow();
    this.resizeEventSubscription = subscribe('resize', this.checkWindow, {
      useRAF: true,
    });
  }

  checkWindow = () => {
    // eslint-disable-next-line
    if (window.matchMedia(MIN_MOBILE_NAV_WIDTH).matches) {
      this.setState({ isNavScrollable: false });
    } else {
      this.setState({ isNavScrollable: true });
    }
  };

  handleItemClick = (e) => {
    const { history } = this.props;
    const { name } = e.target;

    history.push(`/${name}`);
  };

  render() {
    const { isNavScrollable } = this.state;
    const { match } = this.props;
    return (
      <div>
        <h1>Lingo</h1>
        <TabBar scrollable={ isNavScrollable }>
          <TabBarItem
            active={ !match.params.department }
            Component="button"
            onClick={ this.handleItemClick }
            className={ m.bgWhite }
          >
            All
          </TabBarItem>
          <TabBarItem
            name="Supply"
            active={ match.params.department === 'Supply' }
            Component="button"
            onClick={ this.handleItemClick }
            className={ m.bgWhite }
          >
            Supply
          </TabBarItem>
          <TabBarItem
            name="Concierge"
            active={ match.params.department === 'Concierge' }
            Component="button"
            onClick={ this.handleItemClick }
            className={ m.bgWhite }
          >
            Concierge
          </TabBarItem>
          <TabBarItem
            name="Marketing"
            active={ match.params.department === 'Marketing' }
            Component="button"
            onClick={ this.handleItemClick }
            className={ m.bgWhite }
          >
            Marketing
          </TabBarItem>
          <TabBarItem
            name="Service"
            active={ match.params.department === 'Service' }
            Component="button"
            onClick={ this.handleItemClick }
            className={ m.bgWhite }
          >
            Service
          </TabBarItem>
          <TabBarItem
            name="Product"
            active={ match.params.department === 'Product' }
            Component="button"
            onClick={ this.handleItemClick }
            className={ m.bgWhite }
          >
            Product
          </TabBarItem>
        </TabBar>
      </div>
    );
  }
}
