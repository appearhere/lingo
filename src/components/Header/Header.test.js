import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter, Route } from 'react-router';

import Header from './Header';

jest.mock('@appearhere/react-stickynode');
jest.mock('../../utils/terms');

describe('Header component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StaticRouter><Route component={ Header } /></StaticRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
