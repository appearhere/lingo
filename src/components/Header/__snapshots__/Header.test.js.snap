// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `null`;
